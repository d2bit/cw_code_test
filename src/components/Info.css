.Info {
  display: flex;
  flex-flow: column wrap;
  justify-content: space-around;
  align-items: center;
  flex-basis: 70%;
  background-color: #fff;
  border: 1px solid #ccc;
  padding: 1rem;
  margin-bottom: 1rem;
}

.Info-header {
  flex: 1 0 auto;
  border-bottom: 1px solid black;
}

.Info-title {
  font-size: 2rem;
  margin: 0;
}

.Info-price {
  font-size: 2.5rem;
  margin: 2rem 0 1rem;
  color: black;
}

.Info-price-up {
  color: green;
}

.Info-price-down {
  color: red;
}

@media (min-width: 550px) {
  .Info {
    flex-basis: 30%;
  }
}
