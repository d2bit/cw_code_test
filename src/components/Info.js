import React from 'react'
import './Info.css'

class Info extends React.PureComponent {
  priceClassNames = () => {
    const {priceOscilation} = this.props
    let classNames = ['Info-price']

    if (priceOscilation > 0) classNames.push('Info-price-up')
    else if (priceOscilation < 0) classNames.push('Info-price-down')

    return classNames.join(' ')
  }

  render() {
    const {name, symbol, price} = this.props

    return (
      <div className="Info">
        <header className="Info-header">
          <h1 className="Info-title">
            {name} - {symbol}
          </h1>
        </header>
        <h1 className={this.priceClassNames()}>${price}</h1>
      </div>
    )
  }
}

export default Info
