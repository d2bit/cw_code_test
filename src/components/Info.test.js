import React from 'react'
import {render} from 'react-testing-library'
import Info from './Info'

it('renders without crashing', () => {
  const {container} = render(
    <Info name="Bitcoin" symbol="BTC" price={123.45} priceOscilation={0} />,
  )
  expect(container).toMatchSnapshot()
})

it('applies class if priceOscilation is positive', () => {
  const {container} = render(
    <Info name="Bitcoin" symbol="BTC" price={123.45} priceOscilation={0.5} />,
  )
  expect(container).toMatchSnapshot()
})

it('applies class if priceOscilation is negative', () => {
  const {container} = render(
    <Info name="Bitcoin" symbol="BTC" price={123.45} priceOscilation={-0.5} />,
  )
  expect(container).toMatchSnapshot()
})
