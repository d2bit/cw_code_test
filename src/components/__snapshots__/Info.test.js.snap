// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applies class if priceOscilation is negative 1`] = `
<div>
  <div
    class="Info"
  >
    <header
      class="Info-header"
    >
      <h1
        class="Info-title"
      >
        Bitcoin
        - 
        BTC
      </h1>
    </header>
    <h1
      class="Info-price Info-price-down"
    >
      $
      123.45
    </h1>
  </div>
</div>
`;

exports[`applies class if priceOscilation is positive 1`] = `
<div>
  <div
    class="Info"
  >
    <header
      class="Info-header"
    >
      <h1
        class="Info-title"
      >
        Bitcoin
        - 
        BTC
      </h1>
    </header>
    <h1
      class="Info-price Info-price-up"
    >
      $
      123.45
    </h1>
  </div>
</div>
`;

exports[`renders without crashing 1`] = `
<div>
  <div
    class="Info"
  >
    <header
      class="Info-header"
    >
      <h1
        class="Info-title"
      >
        Bitcoin
        - 
        BTC
      </h1>
    </header>
    <h1
      class="Info-price"
    >
      $
      123.45
    </h1>
  </div>
</div>
`;
